### PR TITLE
Added userdata to the SteamNetConnectionStatusChanged callback

### PR DESCRIPTION
**Pull request by yamashi**
This is a fix for #162

### CODE REVIEW

File: include/steam/isteamnetworkingsockets.h [hunk 1]
```diff
@@ -774,6 +774,9 @@ struct SteamNetConnectionStatusChangedCallback_t
 
 	/// Previous state.  (Current state is in m_info.m_eState)
 	ESteamNetworkingConnectionState m_eOldState;
+
+	/// Userdata
+	void* m_userdata;
```
Comment (tycho) [inline include/steam/isteamnetworkingsockets.h:779]: Should be `m_pUserData` for consistency

**Comment by fletcherdvalve** [comment]
Connections have an int64 userdata field already (which is already passed to the connection state changed callback, in the connection info member), so I probably would not want to add a new property.  I would just add a way to get/set the existing userdata property atomically with an option.

I'm still concerned that I'm setting a trap for people who rely on that field in the callback to be correct.  If the userdata is set after a callback has been posted, the callback will not receive the correct userdata.  Probably should add a very stern warning about race conditions in the callback struct, and encourage people to fetch the userdata from the connection instead of using the one in the struct.

**Comment by yamashi** [comment]
Would it be possible to add the userdata as an argument to the function and not an option? This would force the user to pay attention to userdata?

**Comment by fletcherdvalve** [comment]
Superseded ab5ef56aa2cfadf897c8977695436ffb8d8d3ad0

**Pull request closed without merging**